frontend/components/LSP7Panel.tsx

```diff
@@ -58,7 +57,7 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
 
   const fontColor = useColorModeValue('light.black', 'dark.purple.500');
 
-  const tokenAddressDisplay = formatAddress(tokenAddress);
+  const tokenAddressDisplay = formatAddress(tokenData?.address);
```

Comment:
minor: tokenData is not marked optional so no need for `?` (same below for .balance/.metadata/.name)